mod-audit-server/src/main/java/org/folio/dao/acquisition/impl/PieceEventsDaoImpl.java

```diff
@@ -0,0 +1,157 @@
+  @Override
+  public Future<PieceAuditEventCollection> getAuditEventsByPieceId(String pieceId, String sortBy, String sortOrder, int limit, int offset, String tenantId) {
+    LOGGER.debug("getAuditEventsByOrderId:: Trying to retrieve AuditEvent with piece id : {}", pieceId);
+    Promise<RowSet<Row>> promise = Promise.promise();
+    try {
+      LOGGER.debug("formatDBTableName:: Formatting DB Table Name with tenant id : {}", tenantId);
+      String logTable = formatDBTableName(tenantId, TABLE_NAME);
+      String query = format(GET_BY_PIECE_ID_SQL, logTable, logTable, format(ORDER_BY_PATTERN, sortBy, sortOrder));
+      Tuple queryParams = Tuple.of(UUID.fromString(pieceId), limit, offset);
+      pgClientFactory.createInstance(tenantId).selectRead(query, queryParams, promise);
+    } catch (Exception e) {
+      LOGGER.warn("Error getting piece audit events by piece id: {}", pieceId, e);
+      promise.fail(e);
+    }
+    LOGGER.info("getAuditEventsByOrderId:: Retrieved AuditEvent with piece id : {}", pieceId);
+    return promise.future().map(rowSet -> rowSet.rowCount() == 0 ?
+      new PieceAuditEventCollection().withTotalItems(0) :
+      mapRowToListOfPieceEvent(rowSet));
+  }
+
+  @Override
+  public Future<PieceAuditEventCollection> getAuditEventsWithStatusChangesByPieceId(String pieceId, String sortBy, String sortOrder, int limit, int offset, String tenantId) {
+    LOGGER.debug("getAuditEventsByOrderId:: Retrieving AuditEvent with piece id : {}", pieceId);
+    Promise<RowSet<Row>> promise = Promise.promise();
+    try {
+      LOGGER.debug("formatDBTableName:: Formatting DB Table Name with tenant id : {}", tenantId);
+      String logTable = formatDBTableName(tenantId, TABLE_NAME);
+      String query = format(GET_STATUS_CHANGE_HISTORY_BY_PIECE_ID_SQL, logTable, format(ORDER_BY_PATTERN, sortBy, sortOrder));
+      Tuple queryParams = Tuple.of(UUID.fromString(pieceId), limit, offset);
+      pgClientFactory.createInstance(tenantId).selectRead(query, queryParams, promise);
+    } catch (Exception e) {
+      LOGGER.warn("Error getting order audit events by piece id: {}", pieceId, e);
+      promise.fail(e);
+    }
+    LOGGER.info("getAuditEventsByOrderId:: Retrieved AuditEvent with piece id: {}", pieceId);
+    return promise.future().map(rowSet -> rowSet.rowCount() == 0 ? new PieceAuditEventCollection().withTotalItems(0)
+      : mapRowToListOfPieceEvent(rowSet));
+  }
+
+  private PieceAuditEventCollection mapRowToListOfPieceEvent(RowSet<Row> rowSet) {
+    LOGGER.debug("mapRowToListOfOrderEvent:: Mapping row to List of Piece Events");
+    PieceAuditEventCollection pieceAuditEventCollection = new PieceAuditEventCollection();
+    rowSet.iterator().forEachRemaining(row -> {
+      pieceAuditEventCollection.getPieceAuditEvents().add(mapRowToPieceEvent(row));
+      pieceAuditEventCollection.setTotalItems(row.getInteger(TOTAL_RECORDS_FIELD));
+    });
+    LOGGER.info("mapRowToListOfOrderEvent:: Mapped row to List of Piece Events");
```

Comment:
please change this log in all 3 daos for debug

Reply:
changed